src/pkg/services/m365/m365.go

```diff
@@ -98,55 +112,56 @@ type Site struct {
 
 // Sites returns a list of Sites in a specified M365 tenant
 func Sites(ctx context.Context, acct account.Account, errs *fault.Bus) ([]*Site, error) {
-	gc, err := connector.NewGraphConnector(ctx, graph.HTTPClient(graph.NoTimeout()), acct, connector.Sites, errs)
+	sites, err := discovery.Sites(ctx, acct, errs)
 	if err != nil {
 		return nil, clues.Wrap(err, "initializing M365 graph connection")
 	}
 
-	// gc.Sites is a map with keys: SiteURL, values: ID
-	ret := make([]*Site, 0, len(gc.Sites))
-	for k, v := range gc.Sites {
-		ret = append(ret, &Site{
-			WebURL: k,
-			ID:     v,
-		})
+	ret := make([]*Site, 0, len(sites))
+
+	for _, s := range sites {
+		ps, err := parseSite(s)
+		if err != nil {
+			return nil, clues.Wrap(err, "parsing siteable")
+		}
+
+		ret = append(ret, ps)
 	}
 
 	return ret, nil
 }
 
-// SiteURLs returns a list of SharePoint site WebURLs in the specified M365 tenant
-func SiteURLs(ctx context.Context, acct account.Account, errs *fault.Bus) ([]string, error) {
-	gc, err := connector.NewGraphConnector(ctx, graph.HTTPClient(graph.NoTimeout()), acct, connector.Sites, errs)
-	if err != nil {
-		return nil, clues.Wrap(err, "initializing M365 graph connection")
+// parseSite extracts the information from `models.Siteable` we care about
+func parseSite(item models.Siteable) (*Site, error) {
+	s := &Site{
+		ID:     ptr.Val(item.GetId()),
+		WebURL: ptr.Val(item.GetWebUrl()),
 	}
 
-	return gc.GetSiteWebURLs(), nil
+	return s, nil
 }
 
-// SiteIDs returns a list of SharePoint sites IDs in the specified M365 tenant
-func SiteIDs(ctx context.Context, acct account.Account, errs *fault.Bus) ([]string, error) {
-	gc, err := connector.NewGraphConnector(ctx, graph.HTTPClient(graph.NoTimeout()), acct, connector.Sites, errs)
+// SitesMap retrieves all sites in the tenant, and returns two maps: one id-to-webURL,
+// and one webURL-to-id.
+func SitesMap(
+	ctx context.Context,
+	acct account.Account,
+	errs *fault.Bus,
+) (common.IDsNames, error) {
+	sites, err := Sites(ctx, acct, errs)
 	if err != nil {
-		return nil, clues.Wrap(err, "initializing graph connection")
+		return common.IDsNames{}, err
 	}
 
-	return gc.GetSiteIDs(), nil
-}
-
-// parseUser extracts information from `models.Userable` we care about
-func parseUser(item models.Userable) (*User, error) {
-	if item.GetUserPrincipalName() == nil {
-		return nil, clues.New("user missing principal name").
-			With("user_id", clues.Hide(ptr.Val(item.GetId())))
+	ins := common.IDsNames{
+		IDToName: make(map[string]string, len(sites)),
+		NameToID: make(map[string]string, len(sites)),
```

Comment:
this makes me nervous

Reply:
Can you explain your nervousness, please?

Reply:
I'm not really a fan of modifying internal field structs in packages that don't own said structs

Reply:
technically this is construction, not modification.  It'll turn into a real constructor on a later pass.